x-pack/legacy/plugins/apm/server/lib/index_pattern/get_dynamic_index_pattern.ts

```diff
@@ -40,14 +38,20 @@ export const getKueryBarIndexPattern = async ({
 
   const configuredIndices = indexNames.map(name => indicesMap[name]);
 
-  const fields = await indexPatternsFetcher.getFieldsForWildcard({
-    pattern: configuredIndices
-  });
-
-  const pattern: StaticIndexPattern = {
-    fields,
-    title: configuredIndices.join(',')
-  };
-
-  return pattern;
+  try {
```

Comment:
Good question! I'll update with this comment:

```
  // Since `getDynamicIndexPattern` is called in setup_request (and thus by every endpoint)
  // and since `getFieldsForWildcard` will throw if the specified indices don't exist,
  // errors have to be caught to avoid every APM endpoints returning 500 for users 
  // without APM data (would be a bad first time experience)
```